src/main/cpp/CMakeLists.txt

```diff
@@ -94,11 +94,8 @@ include(cmake/Modules/ConfigureCUDA.cmake) # set other CUDA compilation flags
 # ##################################################################################################
 # * dependencies ----------------------------------------------------------------------------------
 
-# find libcu++
-include(${rapids-cmake-dir}/cpm/libcudacxx.cmake)
-
-# find thrust/cub
-include(${CUDF_DIR}/cpp/cmake/thirdparty/get_thrust.cmake)
+# find CCCL
+include(${CUDF_DIR}/cpp/cmake/thirdparty/get_cccl.cmake)
```

Comment:
This can be reverted. We only need the C++ changes for `cuda::proclaim_return_type` in this PR. Then we can update everything (including telling CMake to use CCCL instead of Thrust/libcudacxx) in a follow-up PR.

Reply:
Now with the latest cudf commit we can't compile if don't change this.

Reply:
Correct. If you want to use a newer cudf build, you need the CMake changes. I was hoping we could merge the C++ changes separately but it's fine to combine (nonbreaking) C++ changes with (breaking) CMake changes and submodule updates in this PR for expediency.